### PR TITLE
Fixes #22955 - Update host status on finalize

### PR DESCRIPTION
**Pull request by bastilian**
With this finalizing a job will also update the hosts execution status.

### CODE REVIEW

**Comment by adamruzicka** [comment]
I can't really review this since I was looking into it with @bastilian, @iNecas could you take a look?

**Comment by bastilian** [comment]
Indeed. App code kindly provided by @adamruzicka and works as expected verified with tests by me. :)

File: test/unit/actions/run_host_job_test.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+module ForemanRemoteExecution
+  class RunHostJobTest < ActiveSupport::TestCase
+    include Dynflow::Testing
+
+    subject { create_action(Actions::RemoteExecution::RunHostJob) }
+    let(:host) { FactoryBot.create(:host, :with_execution) }
+
+    before do
+      subject.stubs(:input).returns({
+        host: { id: host.id }
```
[the remark this answers is not in the document]
Comment (iNecas) [reply test/unit/actions/run_host_job_test.rb]: @houndci-bot go home, you're drunk (and not looking at our rubocop config)

**Comment by iNecas** [comment]
Nice work folks. Tested and works almost perfect: waiting for the last comment to get addressed.

**Comment by bastilian** [comment]
@iNecas `FakeScriptRunner#exit_code` now returns a integer.

**Comment by iNecas** [comment]
Thanks, waiting for http://ci.theforeman.org/job/test_plugin_pull_request/4983/ to finish before merging.

**Comment by iNecas** [comment]
Thanks, the pr is now both cool and green as a 🥒. Merging

**Pull request merged**